File: portfolio-fe/src/models/Skill.ts
export interface Skill {
    skillId: string;
    name: string;
    level: string; // e.g., "Beginner", "Intermediate", "Advanced", "Expert"
    category: string; // e.g., "Frontend", "Backend", "Database", "DevOps", "Soft Skill"
    displayOrder?: number; // Optional: for ordering skills
    createdAt?: string; // ISO date string
    updatedAt?: string; // ISO date string
}

export interface SkillRequest {
    name: string;
    level: string;
    category: string;
    displayOrder?: number;
    // imageFile will be handled as File object separately in service/component
} 

File: portfolio-be/src/main/java/com/portfolio/portfolio_backend/datalayer/skills/Skill.java
package com.portfolio.portfolio_backend.datalayer.skills;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "skills")
public class Skill {

    @Id
    private String id; // Internal MongoDB ID
    private String skillId; // Public unique ID
    private String name;
    private String level; // e.g., "Beginner", "Intermediate", "Advanced", "Expert"
    private String category; // e.g., "Frontend", "Backend", "Database", "DevOps", "Soft Skill"
    private Integer displayOrder; // Optional: for ordering skills in a list
    private Instant createdAt;
    private Instant updatedAt;
} 